File: leetcode657.cpp
class Solution {
public:
    bool judgeCircle(string moves) {
      int x = 0,y = 0;
      int i = 0;
      while(i < moves.length()){
        char ch = moves[i ++];
        if(ch == 'U') y ++;
        else if(ch == 'D') y --;
        else if(ch == 'L') x --;
        else x ++;
      }
      return x == 0 && y == 0;
    }
};

File: leetcode657.py
class Solution:
    def judgeCircle(self, moves: str) -> bool:
        from collections import Counter
        ans = Counter(moves)
        return (abs(ans.get('U',0) - ans.get('D',0))+ abs(ans.get('R',0) - ans.get('L',0))) == 0
